rclpy/test/test_node.py

```diff
@@ -838,6 +842,127 @@ def test_node_set_parameters_rejection(self):
         self.assertIsInstance(result[0], SetParametersResult)
         self.assertFalse(result[0].successful)
 
+    def test_node_set_parameters_rejection_list(self):
+        # Declare a new parameters and set a list of callbacks so that it's rejected when set.
+        reject_list_parameter_tuple = [
+            ('reject', True, ParameterDescriptor()),
+            ('non_reject', True, ParameterDescriptor()),
+            ('non_reject', True, ParameterDescriptor())
+        ]
+        self.node.declare_parameters('', reject_list_parameter_tuple)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback)
+        self.node.add_on_set_parameters_callback(self.reject_parameter_callback_1)
+
+        result = self.node.set_parameters(
+            [
+                Parameter(
+                    name=reject_list_parameter_tuple[0][0],
+                    value=reject_list_parameter_tuple[0][1]
+                ),
+                Parameter(
+                    name=reject_list_parameter_tuple[1][0],
+                    value=reject_list_parameter_tuple[1][1]
+                ),
+                Parameter(
+                    name=reject_list_parameter_tuple[2][0],
+                    value=reject_list_parameter_tuple[2][1]
+                )
+            ]
+        )
+        self.assertIsInstance(result, list)
+        self.assertIsInstance(result[0], SetParametersResult)
+        self.assertFalse(result[0].successful)
+        # Testing with setting new parameter with successful result with
+        # Previously declared function.
+        self.test_node_set_parameters()
```

Comment:
I don't think it's a good idea to call another test from within a test. I think it will be easier to maintain the test code if they are kept separate.

Instead, I think we should assert that the non-rejected parameters were set successfully, for example:

```py
self.assertIsInstance(result[1], SetParametersResult)
self.assertTrue(result[1].successful)
self.assertIsInstance(result[2], SetParametersResult)
self.assertTrue(result[2].successful)
```